src/nix/run.cc

```diff
@@ -121,7 +124,7 @@ struct CmdShell : InstallablesCommand, MixEnvironment
 
             auto propPath = store->printStorePath(path) + "/nix-support/propagated-user-env-packages";
             if (accessor->stat(propPath).type == FSAccessor::tRegular) {
-                for (auto & p : tokenizeString<Paths>(readFile(propPath)))
+                for (auto & p : tokenizeString<Paths>(readFile(store->toRealPath(propPath))))
```

Comment:
Since we already have an `accessor`, can't we do:

```suggestion
                for (auto & p : tokenizeString<Paths>(accessor->readFile(propPath)))
```

Reply:
Good point, tested and applied.